lib/cn/csched_sp3_work.c

```diff
@@ -600,7 +681,7 @@ sp3_work(
     if (cn_node_isleaf(tn)) {
         switch (wtype) {
         case wtype_split:
-            n_kvsets = sp3_work_wtype_split(spn, thresh, &mark, &action, &rule);
+            n_kvsets = sp3_work_wtype_split(spn, thresh, &mark, &action, &rule, &(*wp)->cw_resched);
             break;
```

Comment:
```
n_kvsets = sp3_work_wtype_split(spn, thresh, &mark, &action, &rule);
wp->cw_resched = kvsets != 0;
```

Reply:
Right, I had that once.  But if sp3_work_wtype_split() decides we shouldn't split then csched will just keep asking...

Reply:
I reverted it for now...